src/metrics/listener.ts

```diff
@@ -138,9 +142,20 @@ export class MetricsListener {
     if (this.isExtensionRunning) {
       const isMetricTimeValid = Date.parse(metricTime.toString()) > 0;
       if (isMetricTimeValid) {
+        const dateCeiling = new Date(Date.now() - 4 * 60 * 60 * 1000); // 4 hours ago
+        if (dateCeiling > metricTime) {
+          logWarning(
+            "The timestamp provided is too old to be sent to the Datadog API. Please provide a timestamp within the last 4 hours.",
+          );
```

Comment:
Would it be possible to get this text to be more specific? Something like `Metric {name} ..." in case a user is sending thousands of metrics so its easier to debug?

Reply:
Also, is there a test for us to check that a metric is indeed ignored?

Reply:
yessir, will add these 👍